src/Microsoft.IdentityModel.Tokens/Validation/LifetimeValidationResult.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+#nullable enable
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Contains the result of validating the lifetime of a <see cref="SecurityToken"/>.
+    /// The <see cref="TokenValidationResult"/> contains a collection of <see cref="ValidationResult"/> for each step in the token validation.
+    /// </summary>
+    internal class LifetimeValidationResult : ValidationResult
+    {
+        private Exception? _exception;
+
+        /// <summary>
+        /// Creates an instance of <see cref="LifetimeValidationResult"/>
+        /// </summary>
+        /// <paramref name="notBefore"/> is the date from which the token that was validated successfully is valid.
+        /// <paramref name="expires"/> is the expiration date for the token that was validated successfully.
+        public LifetimeValidationResult(DateTime? notBefore, DateTime? expires)
```

Comment:
I guess these could be null when validated from a delegate?

Reply:
Added a test to highlight the case where these can be null. Mentioned on the signing key PR about how there are flags to disable validations that make these methods return positive under potential null values.